include/shark/Algorithms/DirectSearch/CMA.h

```diff
@@ -204,6 +204,20 @@ class CMA : public AbstractSingleObjectiveOptimizer<RealVector >
 		return m_numEvaluations;
 	}
 
+	///\breif Return the population form the previous step
+	std::vector<SolutionType> population() const {
+		return m_population;
+	}
+
+	///\brief Returns the best solution of the previous step
+	SolutionType best() const {
+		return m_best;
+	}
+
+	///\brief Returns the best ever discovered solution
+	SolutionType bestEver() const {
```

Comment:
If we want that, this should also go in the base class. But I am not sure that we want that. Often, what is considered the "all-time-best" solution would depend on a different metric, e.g. classification error of the model.

Reply:
I see your point. I will remove these in a PR update. If the `best` ever is to be tracked, one might as well do this manually as the `best` is exposed per iteration.